pinot-core/src/main/java/org/apache/pinot/core/indexsegment/mutable/MutableSegmentImpl.java

```diff
@@ -379,6 +393,25 @@ private void addInvertedIndex(int docId, Map<String, Object> dictIdMap) {
     }
   }
 
+  /**
+   * Check if the row has any null fields and update the
+   * column presence vectors accordingly
+   * @param row specifies row being ingested
+   * @param docId specified docId for this row
+   */
+  private void addPresenceVector(GenericRow row, int docId) {
+    if (!row.hasNullValues()) {
+      return;
+    }
+
+    for (FieldSpec fieldSpec : _schema.getAllFieldSpecs()) {
```

Comment:
Iterate over `row.getNullValueFields()` and set presence vector

Reply:
Done